gc/base/SlotObject.hpp

```diff
@@ -48,24 +51,44 @@ class GC_SlotObject
 	MMINLINE omrobjectptr_t
 	convertPointerFromToken(fomrobject_t token)
 	{
+		uintptr_t value = (uintptr_t)token;
 #if defined (OMR_GC_COMPRESSED_POINTERS)
-		return (omrobjectptr_t)((uintptr_t)token << _compressedPointersShift);
-#else
-		return (omrobjectptr_t)token;
-#endif
+		if (compressObjectReferences()) {
+			value <<= _compressedPointersShift;
```

Comment:
did you consider always shifting, but with 0 for nonCR?

Reply:
The problem is that the _compressedPointerShift field only exists if the compressed ifdef is true. In the future, if we stop ifdeffing the fields, I think the code would look better as you suggest.